contrib/in-memory-accumulo/src/main/java/datawave/accumulo/inmemory/InMemoryBatchScanner.java

```diff
@@ -97,21 +97,27 @@ public void setRanges(Collection<Range> ranges) {
         // Note the key assumption here is that the ranges are processed in order (see IteratorChain
         // used above) and that the ranges are non-overlapping (see Range.mergeOverlapping() used
         // above).
-        List<Range> newRanges = new ArrayList<>();
-        boolean found = false;
-        for (Range range : ranges) {
-            if (found) {
-                newRanges.add(range);
-            } else if (range.contains(lastKey)) {
-                found = true;
-                Range newRange = new Range(lastKey, false, range.getEndKey(), range.isEndKeyInclusive());
-                newRanges.add(newRange);
+        if (lastKey != null) {
+            List<Range> newRanges = new ArrayList<>();
+            boolean found = false;
+            for (Range range : ranges) {
+                if (found) {
+                    newRanges.add(range);
+                } else if (range.contains(lastKey)) {
+                    found = true;
+                    Range newRange = new Range(lastKey, false, range.getEndKey(), range.isEndKeyInclusive());
+                    newRanges.add(newRange);
+                }
             }
+            if (!ranges.isEmpty() && newRanges.isEmpty()) {
```

Comment:
Can ranges every actually be empty initially?

Reply:
this is more a sanity check than anything else. In early testing due to code errors it was possible to get an empty range here